tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -1619,6 +1619,107 @@ def leaderboard(self, data=None, extra_info=False, extra_metrics=None, only_pare
+        for m in models:
+            predict_proba_dict[m] = predictor.predict_proba(data, model=m)
+        ```
+
+        Note that this will generally be much faster than calling `self.predict_proba` separately for each model
+        because this method leverages the model dependency graph to avoid redundant computation.
+
+        Parameters
+        ----------
+        data : DataFrame, default = None
+            The data to predict on.
+            If None:
+                If self.trainer.has_val, the validation data is used.
+                Else, the out-of-fold prediction probabilities are used.
+        models : List[str], default = None
+            The list of models to get predictions for.
+            If None, all models that can infer are used.
+        as_pandas : bool, default = True
+            Whether to return the output of each model as a pandas object (True) or numpy array (False).
+            Pandas object is a DataFrame if this is a multiclass problem or `as_multiclass=True`, otherwise it is a Series.
+            If the output is a DataFrame, the column order will be equivalent to `predictor.class_labels`.
+        as_multiclass : bool, default = True
+            Whether to return binary classification probabilities as if they were for multiclass classification.
+                Output will contain two columns, and if `as_pandas=True`, the column names will correspond to the binary class labels.
+                The columns will be the same order as `predictor.class_labels`.
+            If False, output will contain only 1 column for the positive class (get positive_class name via `predictor.positive_class`).
+            Only impacts output for binary classification problems.
+        transform_features : bool, default = True
+            If True, preprocesses data before predicting with models.
+            If False, skips global feature preprocessing.
+                This is useful to save on inference time if you have already called `data = predictor.transform_features(data)`.
+        inverse_transform : bool, default = True
+            If True, will return prediction probabilities in the original format.
+            If False (advanced), will return prediction probabilities in AutoGluon's internal format.
+
+        Returns
+        -------
+        Dictionary with model names as keys and model prediction probabilities as values.
+        """
+        return self._learner.predict_proba_dict(X=data,
+                                                models=models,
+                                                as_pandas=as_pandas,
+                                                as_multiclass=as_multiclass,
+                                                transform_features=transform_features,
+                                                inverse_transform=inverse_transform)
+
+    def predict_dict(self, data=None, models=None, as_pandas=True, transform_features=True, inverse_transform=True) -> dict:
```

Comment:
nit: `... -> Dict[str, DataFrame]` + add parameter types

Reply:
added, note type hint for data and output will wait for future PR due to multiple types support